scipy/optimize/minpack.py

```diff
@@ -679,6 +679,9 @@ def curve_fit(f, xdata, ydata, p0=None, sigma=None, absolute_sigma=False,
         if ier not in [1, 2, 3, 4]:
             raise RuntimeError("Optimal parameters not found: " + errmsg)
     else:
+        # Rename maxfev (leastsq) to max_nfev (least_squares).
+        kwargs['max_nfev'] = kwargs.pop('maxfev', None)
```

Comment:
Looks like 'max_nfev' will be overwritten (often by None) if it was passed. I think a precondition `if 'max_nfev' not in kwargs` is necessary.